src/handlers.cpp

```diff
@@ -2543,11 +2545,52 @@ void PushProfileTask::on_get_irs_failure(CassandraStore::Operation* op,
 
 void PushProfileTask::update_reg_data()
 {
+  TRC_DEBUG("PPR is valid");
+  HTTPCode ret_code = 0;
   if (!_ims_sub_present || !is_first_irs())
   {
     _impus = _irs_impus;
   }
 
+  if (_ims_sub_present && is_first_irs())
+  {
+    find_impus_to_delete();
+
+    TRC_DEBUG("Going to inform sprout of user data");
+
+    ret_code = _cfg->sprout_conn->change_associated_identities(_default_public_id,
+	       			 	   	               _ims_subscription,
+							       this->trail());
+    switch (ret_code)
```

Comment:
I think this switch statement is overkill. This would be better as:
```
if (ret_code == HTTP_OK)
{
  TRC statement for success
}
else
{
  TRC statement for error
  send_ppa
  delete
  return
}
```